plugins/sonarqube/src/api/SonarQubeClient.ts

```diff
@@ -34,20 +35,37 @@ export class SonarQubeClient implements SonarQubeApi {
     this.baseUrl = baseUrl.endsWith('/') ? baseUrl : `${baseUrl}/`;
   }
 
-  private async callApi<T>(path: string): Promise<T | undefined> {
+  private async callApi<T>(
+    path: string,
+    query?: { [key in string]: any },
+  ): Promise<T | undefined> {
+    const queryString = query ? `?${qs.stringify(query)}` : '';
     const apiUrl = `${await this.discoveryApi.getBaseUrl('proxy')}/sonarqube`;
-    const response = await fetch(`${apiUrl}/${path}`);
+    const response = await fetch(`${apiUrl}/${path}${queryString}`);
     if (response.status === 200) {
       return (await response.json()) as T;
     }
     return undefined;
   }
 
   private async getSupportedMetrics(): Promise<string[]> {
-    const result = await this.callApi<{ metrics: Array<{ key: string }> }>(
-      'metrics/search',
-    );
-    return result?.metrics?.map(m => m.key) ?? [];
+    const metrics: string[] = [];
+    let pageSize: number | undefined;
+    let nextPage: number = 1;
+
+    do {
+      const result = await this.callApi<{
+        metrics: Array<{ key: string }>;
+        total: number;
+      }>('metrics/search', { ps: 500, p: nextPage });
+
+      metrics.push(...(result?.metrics?.map(m => m.key) ?? []));
+
+      pageSize = result?.total;
```

Comment:
Logic here with `result === undefined` kinda works, but could be a bit more explicit :p

Reply:
Refactored it to be more explicit. Also realized that my old code interpreted the SonarQube API wrongly 😅